### PR TITLE
Replace VB type check and cast with TryCast and null check

### PR DESCRIPTION
**Pull request by drewnoakes**

### CODE REVIEW

**Comment by davkean** [comment]
Were these automated?

**Comment by drewnoakes** [comment]
Yes, by R#. Not flagged by Roslyn or analyzers.

**Comment by davkean** [comment]
Were the changes automated? Just wondering how much of this I need to review.

**Comment by drewnoakes** [comment]
Yes they were automated. R# added parens, which I removed by hand.

File: src/Microsoft.VisualStudio.Editors/DesignerFramework/GenericComponentSerializationStore.vb [hunk 1]
```diff
@@ -350,11 +350,9 @@ Namespace Microsoft.VisualStudio.Editors.DesignerFramework
...
                 ' If it is an IComponent, we'll try to get its name from 
                 ' its site
-                If TypeOf Value Is IComponent Then
-                    Dim comp As IComponent = DirectCast(Value, IComponent)
-                    If comp.Site IsNot Nothing Then
-                        _objectName = comp.Site.Name
-                    End If
+                Dim comp = TryCast(Value, IComponent)
+                If comp IsNot Nothing AndAlso comp.Site IsNot Nothing Then
+                    _objectName = comp.Site.Name
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/DesignerFramework/GenericComponentSerializationStore.vb]: I manually introduced `AndAlso` here.

File: src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb [hunk 2]
```diff
@@ -761,12 +761,13 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
...
-                                If (TypeOf obj Is String) Then
-
+                                Dim o = TryCast(obj, String)
+                                If o IsNot Nothing Then
                                     ' now we finally have the default-namespace
                                     '
-                                    rootNamespace = CType(obj, String)
+                                    rootNamespace = o
+                                Else
+                                    Debug.Fail("DefaultNamespace didn't return a string?")
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb]: I manually moved this `Debug` call and changed it to `Fail`.

File: src/Microsoft.VisualStudio.Editors/XmlIntellisense/XmlIntellisense.vb [hunk 3]
```diff
@@ -769,37 +770,34 @@ Namespace Microsoft.VisualStudio.Editors.XmlIntellisense
             Dim ExistingMember As XmlIntellisenseMember = Nothing
 
             ' If the element's type has complex content,
-            If TypeOf Element.ElementSchemaType Is XmlSchemaComplexType Then
-                Dim ContentType As XmlSchemaComplexType = DirectCast(Element.ElementSchemaType, XmlSchemaComplexType)
+            Dim ContentType = TryCast(Element.ElementSchemaType, XmlSchemaComplexType)
+            If ContentType IsNot Nothing Then
 
-                If ContentType IsNot Nothing Then
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/XmlIntellisense/XmlIntellisense.vb]: The diff below is mostly whitespace due to reducing nesting level.

**Pull request merged**